uaclient/cli.py

```diff
@@ -97,6 +98,38 @@ def new_f(args, cfg):
     return new_f
 
 
+def daemon_file_exists():
+    return os.path.exists(defaults.DEFAULT_UPGRADE_CONTRACT_FLAG_FILE)
+
+
+def get_running_services():
+    running_services, _ = util.subp(["ps", "-aux"])
+    return running_services
+
+
+def assert_contract_upgrade_not_running(f):
+    """Decorator asserting contract upgrade is not running."""
+
+    @wraps(f)
+    def new_f(args, cfg):
+        if daemon_file_exists():
+            running_services = get_running_services()
+            daemon_name = defaults.DAEMON_UPGRADE_CONTRACT_NAME
+
+            if running_services.find(daemon_name) == -1:
```

Comment:
this could be nicely expressed it a more pythonic way as the following
```suggestion
            if daemon_name in running_services:
```